sourced/ml/repo2/coocc.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import os
+from operator import add
+from uuid import uuid4
+from pyspark.sql import column
+from pyspark.sql.functions import size, col
+
+from modelforge.meta import generate_meta
+from modelforge.model import merge_strings, write_model
+from sourced.ml.engine import create_engine, get_tokens
+from sourced.ml.repo2.base import UastExtractor, Transformer, Cacher, UastDeserializer
+from sourced.ml.repo2.token_map import TokenMapTransformer
+from sourced.ml.token_parser import TokenParser
+
+import itertools
+
+from sourced.ml.bblfsh_meta import Roles, Node
+
+
+class SaveASDFCooccModel(Transformer):
```

Comment:
Remove "asdf" - it is an implementation detail. Change the rest: `CooccModelSaver` - class names must be nouns.